### PR TITLE
[FIX] web: group (tag) size adaptation

### PR DESCRIPTION
**Pull request by thsh-odoo**
**Before this PR:**

Suppose there are 2 fields in a group tag and in that tag one field is invisible so the group tag will make the tr for both fields due to this the view is broken.

**After this PR:**

Now the group tag will not make tr for the invisible field.

Task-3179173

### CODE REVIEW

**Comment by amdi-odoo** [comment]
Hello @thsh-odoo :slightly_smiling_face: 

Globally, it is a bad practice to change the web module views which are used everywhere in Odoo just to fix a display issue in a certain module (here CRM).

----------------------------------------------------------------------------------------------------------------------------------------------

From Odoo 16.0, the forms are using groups with a grid system to display the fields: https://tinyurl.com/235952h9

So to fix the display in 16.0, just tell the xml that the field is taking the 2 columns of the grid by setting a `colspan="2"` to the field in the `crm_lead_to_opportunity_views.xml` like so:
```xml
<field name="duplicated_lead_ids" colspan="2" nolabel="1">
```
Here is how the grid will render: https://tinyurl.com/26zhxek7

In 14.0, there is no grid system, but setting a `colspan="2"` will also fix the issue as it will automatically change the style from `width: 50%` to  `width: 100%` : 
image


In short, adding `colspan="2"` on the `duplicated_lead_ids` field is fixing the table display for all the Odoo versions :slightly_smiling_face: 
Could you change your fix to this ? :slightly_smiling_face: 
Thank you!

**Comment by thsh-odoo** [comment]
Hello @amdi-odoo 

The main reason to do this change is that group tag was making an unnecessary <tr>  for an invisible field
unnecessary_tr and due to that view breakes and with " **if(!child.attrs.invisible){  }**  " 
condition will not allow an invisible field to make it unnecessary **tr** and i have also spotted same issue in 
employee module https://www.awesomescreenshot.com/image/39090119?key=c82c98381e8fa88a19c3fb540dda65bb 

So what are your thoughts on this 
Thanks

**Comment by amdi-odoo** [comment]
Hello @thsh-odoo :slightly_smiling_face: 
Oh ok got it sorry! :smiling_face_with_tear: 

I agree this useless tr shouldn't be there :thinking: I think we have 2 options : 
1) Fix the display by simply set a `colspan="2"` on the field, but we still have the unnecessary invisible tr
2) We could get rid of this unnecessary tr in the web module but this might be a more risky approach since we're not 100% sure it will not mess up other form views.

For 1), a single PR in 14.0 adding the `colspan="2"` to the fields will be enough to fix the view from 14.0 to master.

For 2), 2 PR will be needed. One in 14.0 (->15.2) to add the web js fix and one in 16.0 (-> master) to add the colspan to the fields and fix the grid display.

Also if we choose 2), simply set this condition at the start of the foreach loop might be more concise and will reduce the commit diff : 
```js
node.children.forEach(function (child) {
     if (child.attrs.invisible) {
            return;
      }
      ...
}
```
I'm not sure if 1 or 2 is better here so I'll just ask @awa-odoo if he has time to give his opinion on this one :sweat_smile: 

Thank you for your time!

**Comment by awa-odoo** [comment]
Hello @amdi-odoo @thsh-odoo 

I was pinged so here I am.
It indeed seems risky to change the JS code of the form_renderer in v14 for a display issue.

It's a central part of the framework and could have deeper repercussions.
Especially in v14 where we normally only fix tracebacks or issues that prevent people from working.

If you really think it's important, maybe you could see with the JS team directly.

Hope that helps.

Cheers!

**Comment by amdi-odoo** [comment]
Hello @thsh-odoo :slightly_smiling_face: 

I agree with awa that it's kind of risky to modify the web js in 14.0 for the repercussions so I think it might be wiser to go with option 1.

Can you check **for each Odoo version** if setting a `colspan="2"` on the field is fixing the table display ? If so, then just a PR in 14.0 with the colspan will do :slightly_smiling_face: 

Sorry for the long discussion here, I wasn't sure myself on what to do.
Hope it helps! 

Thanks and cheers!

**Comment by C3POdoo** [comment]
Dear @thsh-odoo,

Thank you for your contribution but the version 14.0 is no longer supported.
We only support the last 3 stable versions so no longer accepts patches into this branch.

We apology if we could not look at your request in time.
If the contribution still makes sense for the upper version, please let us know and do not hesitate to recreate one for the recent versions. We will try to check it as soon as possible.

_This is an automated message._
